common/rpc/interceptor/redirection.go

```diff
@@ -296,11 +296,16 @@ func (i *Redirection) AfterCall(
 	metricsHandler metrics.Handler,
 	startTime time.Time,
 	clusterName string,
+	namespaceName string,
 	retError error,
 ) {
 	metricsHandler = metricsHandler.WithTags(metrics.TargetClusterTag(clusterName))
-	metrics.ClientRedirectionRequests.With(metricsHandler).Record(1)
 	metrics.ClientRedirectionLatency.With(metricsHandler).Record(i.timeSource.Now().Sub(startTime))
+
+	if len(namespaceName) != 0 {
+		metricsHandler = metricsHandler.WithTags(metrics.NamespaceTag(namespaceName))
+	}
+	metrics.ClientRedirectionRequests.With(metricsHandler).Record(1)
```

Comment:
Can we remove the counter and depends on the latency metric count?

Reply:
we need counter for namespace label. latency + namespace is not recommended.

Reply:
do we also need the namespace tag on error count?

Reply:
I think it wouldn't hurt?